torch_np/_dtypes.py

```diff
@@ -343,21 +343,69 @@ def __setstate__(self, value):
 # ### Defaults and dtype discovery
 
 
-def default_int_type():
-    return dtype(_dtypes_impl.default_int_dtype)
-
-
-def default_float_type():
-    return dtype(_dtypes_impl.default_float_dtype)
-
-
-def default_complex_type():
-    return dtype(_dtypes_impl.default_complex_dtype)
-
-
-def get_default_dtype_for(dtyp):
-    torch_dtype = dtype(dtyp).torch_dtype
-    return _dtypes_impl.get_default_type_for(torch_dtype)
+def set_default_dtype(fp_dtype="numpy", int_dtype="numpy"):
+    """Set the (global) defaults for fp, complex, and int dtypes.
+
+    The complex dtype is inferred from the float (fp) dtype. It has
+    a width at least twice the width of the float dtype,
+    i.e., it's complex128 for float64 and complex64 for float32.
+
+    Parameters
+    ----------
+    fp_dtype
+        Allowed values are "numpy", "pytorch" or dtype_like things which
+        can be converted into a DType instance.
+        Default is "numpy" (i.e. float64).
+    int_dtype
+        Allowed values are "numpy", "pytorch" or dtype_like things which
+        can be converted into a DType instance.
+        Default is "numpy" (i.e. int64).
+
+    Returns
+    -------
+    The old default dtype state: a namedtuple with attributes ``float_dtype``,
+    ``complex_dtypes`` and ``int_dtype``. These attributes store *pytorch*
+    dtypes.
+
+    Notes
+    ------------
+    This functions has a side effect: it sets the global state with the provided dtypes.
+
+    The complex dtype has bit width of at least twice the width of the float
+    dtype, i.e. it's complex128 for float64 and complex64 for float32.
```

Comment:
In the end this was repeated in two places. We can probably delete this bit in any other PR